### PR TITLE
E0d/update and reorg

### PR DESCRIPTION
**Pull request by e0d**
@feanil @benpatterson after pushing this to master and reverting it, here are the changes I propose to upgrade requests and make some other improvements.

Currently I'm seeing one test failure.

### CODE REVIEW

**Comment by e0d** [comment]
Had missed @benpatterson travis PR, but have pulled in his requirements changes and updated to use nose.

Still seeing an error that is swallowed by nose though.

```
usage: xqueue_watcher [-h] -d CONFIG_ROOT
xqueue_watcher: error: argument -d/--config_root is required
```

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	@echo 'Makefile for the xqueue-watcher                                                   		'
+	@echo '                                                                                     		'
+	@echo 'Usage:                                                                               		'
+	@echo '    make requirements                 install requirements for local development     		'
+	@echo '    make test                         install requirements for local development     		'
+	@echo '    make clean                        delete generated byte code and coverage reports		'
+	@echo '                                                                                     		'
+
+requirements: 
+	pip install -qr requirements/production.txt --exists-action w
```
Comment (benpatterson) [inline Makefile]: should this be -q -r ? If -qr works, we should make the next `pip` statement under test.requirements consistent.

**Comment by benpatterson** [comment]
@e0d when you say the error is swallowed by nose what do you mean? For example, nose will still show you stderr, even if it's part of some self.assertRaises in a test.

**Comment by e0d** [comment]
Meaning I see the output of the error, but nose reports no failures.

From unittest

```
Ran 25 tests in 8.038s

FAILED (failures=1)
(xqwatcher)[edward@alexis xqueue-watcher]$ python -m unittest discover ./tests
```

From nose

```
nosetests --with-coverage --cover-package=xqueue_watcher --cover-xml
git commit -m 'changing to use nose'
(xqwatcher)[edward@alexis xqueue-watcher]$ nosetests --with-coverage --cover-package=xqueue_watcher --cover-xml
...........usage: xqueue_watcher [-h] -d CONFIG_ROOT
xqueue_watcher: error: argument -d/--config_root is required
..............
Name                             Stmts   Miss  Cover   Missing
--------------------------------------------------------------
xqueue_watcher.py                    0      0   100%   
xqueue_watcher/client.py           126     10    92%   24, 36-38, 88, 158-167
xqueue_watcher/grader.py            90      0   100%   
xqueue_watcher/jailedgrader.py     128     30    77%   32, 64, 97-98, 109-111, 131-132, 140, 145-148, 161-163, 173, 177-180, 189-191, 197-199, 205-216, 228
xqueue_watcher/manager.py          109      9    92%   47, 78, 84-87, 161-163
--------------------------------------------------------------
TOTAL                              453     49    89%   
----------------------------------------------------------------------
Ran 25 tests in 6.870s

OK
```

**Comment by benpatterson** [comment]
@e0d that is no problem. That's happening [here](https://github.com/edx/xqueue-watcher/blob/29bb836002ca6a37aecd8563b2347044a6b19737/tests/test_manager.py#L152). It's just nose not-swallowing stderr in this situation. diff-cover has a ton of these because it has CLI-related tests.

**Comment by benpatterson** [comment]
BTW :+1: from me

**Comment by feanil** [comment]
:+1:

**Pull request merged**